app/src/activity.c

```diff
@@ -47,19 +48,21 @@ enum zmk_activity_state zmk_activity_get_state() { return activity_state; }
 int activity_event_listener(const zmk_event_t *eh) {
     activity_last_uptime = k_uptime_get();
 
-    return set_state(ZMK_ACTIVITY_ACTIVE);
+    return activity_set_state(ZMK_ACTIVITY_ACTIVE);
 }
 
 void activity_work_handler(struct k_work *work) {
     int32_t current = k_uptime_get();
     int32_t inactive_time = current - activity_last_uptime;
 #if IS_ENABLED(CONFIG_ZMK_SLEEP)
-    if (inactive_time > MAX_SLEEP_MS) {
-        set_state(ZMK_ACTIVITY_SLEEP);
+    // Second half of || statement is to ensure a user using &sleep doesn't lose sleep state when
+    // the idle or sleep timers expire
+    if (inactive_time > MAX_SLEEP_MS || activity_state == ZMK_ACTIVITY_SLEEP) {
```

Comment:
I'm wondering if this handler is even called when ZMK is in deep sleep. I'd imagine it shouldn't, but if it does, why don't we just clear the timer? Seems unnecessary to run this while we're in sleep.

Reply:
I didn't see a great spot to reset the timer and if the timer code does run at all in sleep mode... we're still going to need the 2nd half of the condition to ensure it's never going to wake up when asleep.

Reply:
What I'm thinking is this handler shouldn't even be called once you're in deep sleep. Have you tested it without this additional conditional? From my understanding when we go into deep sleep, timer-based handlers are no longer run since we don't set up any timer wakeups and therefore don't have any timers running.